app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -79,12 +79,11 @@ public function createOrder($reference, $sessionQuoteId = null, $isAjaxRequest =
                         $immutableQuote->getParentQuoteId() )
                 );
             } else if (!$parentQuote->getIsActive() ) {
-                throw new Exception(
-                    Mage::helper('boltpay')->__("The parent quote %s for immutable quote %s is currently being processed or has been processed for order #%s. Check quote %s for details.",
-                        $parentQuote->getId(), $immutableQuote->getId(), $parentQuote->getReservedOrderId(), $parentQuote->getParentQuoteId() )
-                );
+                $parentQuoteBoltReference = $parentQuote->getBoltReference();
+                throw new Bolt_Boltpay_DuplicateTransitionException($parentQuoteBoltReference, Mage::helper('boltpay')->__("The parent quote %s is currently being processed or has been processed by bolt transaction %s.",
```

Comment:
We discussed this before in the PR https://github.com/BoltApp/bolt-magento1/pull/144/commits/27cfe120fb657cc944cfa320fa45ac66b6f84de2 and https://github.com/BoltApp/bolt-magento1/pull/144/commits/1754dd5508c87a7ef2017a081c944d5c6ff4104a

Reply:
I think we've discussed about response code but not about control flow? 

proposal here is if it's `$reference == $parentQuote->getBoltReference()`, we can simply return here instead of throwing exception - caller can treat that case as same as successful order creation case.

If this makes control flow complex, I'm fine with exception though.